src/BeadingStrategy/RedistributeBeadingStrategy.cpp

```diff
@@ -0,0 +1,72 @@
+//Copyright (c) 2020 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "RedistributeBeadingStrategy.h"
+
+#include <algorithm>
+
+namespace cura
+{
+    BeadingStrategy::Beading RedistributeBeadingStrategy::compute(coord_t thickness, coord_t bead_count) const
+    {
+        Beading ret = parent->compute(thickness, bead_count);
+
+        // Actual count and thickness as represented by extant walls. Don't count any potential zero-width 'signalling' walls.
+        bead_count = std::count_if(ret.bead_widths.begin(), ret.bead_widths.end(), [](const coord_t& width) { return width > 0; });
+        thickness -= ret.left_over;
+
+        // Early out when the only walls are outer, the parent can have been trusted to handle it.
+        if (bead_count < 3)
+        {
+            return ret;
+        }
+
+        // Calculate the factors with which to multiply the outer and inner walls:
+        // (There may be a way to do this with all-integer math, but the speedup won't probably be that much and this will be easier to maintain).
+        const float current_total_outer_walls_width = ret.bead_widths.front() + ret.bead_widths.back();
```

Comment:
all of these float types introduces a lot of implicit and narrowing conversions, maybe reconsider the types